core/src/pow/types.rs

```diff
@@ -389,7 +389,7 @@ impl Proof {
 	}
 }
 
-fn extract_bits(bits: &Vec<u8>, bit_start: usize, bit_count: usize, read_from: usize) -> u64 {
+fn extract_bits(bits: &[u8], bit_start: usize, bit_count: usize, read_from: usize) -> u64 {
```

Comment:
:+1:

Reply:
Also there is several of place we could do something similar that I did not include in this PR.